### PR TITLE
Add Regulations Z1) and Z3) to the Create Competition form.

### PR DESCRIPTION
**Pull request by AlbertoPdRF**
This is another step towards #2472.

Screenshots:

- Create Competition form:

image

- Email:

image

### CODE REVIEW

File: WcaOnRails/db/migrate/20180831164420_add_z1_and_z3_columns_to_competitions.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class AddZ1AndZ3ColumnsToCompetitions < ActiveRecord::Migration[5.2]
+  def change
+    add_column :Competitions, :regulation_z1, :boolean, null: true, default: nil
+    add_column :Competitions, :regulation_z1_reason, :string, null: true, default: nil
+
+    add_column :Competitions, :regulation_z3, :boolean, null: true, default: nil
+    add_column :Competitions, :regulation_z3_reason, :string, null: true, default: nil
```
Comment (jonatanklosko) [inline WcaOnRails/db/migrate/20180831164420_add_z1_and_z3_columns_to_competitions.rb]: Shouldn't this be `text` (or perhaps `mediumtext`) like the `competitor_limit_reason` column? I don't know how long these are expected to be, just want to ensure `string` is intended =)

Comment (AlbertoPdRF) [reply WcaOnRails/db/migrate/20180831164420_add_z1_and_z3_columns_to_competitions.rb]: Yes, it will be better to have it like text.

**Comment by AlbertoPdRF** [comment]
I addressed @jonatanklosko's concern!

**Comment by AlbertoPdRF** [comment]
The reason is that we don't know if past competitions applied this Regulation or not, so setting it to `NULL` is the way to say "we don't know".

I can resolve the conflict tomorrow!

File: WcaOnRails/db/migrate/20180831164420_add_z1_and_z3_columns_to_competitions.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class AddZ1AndZ3ColumnsToCompetitions < ActiveRecord::Migration[5.2]
+  def change
+    add_column :Competitions, :regulation_z1, :boolean, null: true, default: nil
```
[the remark this answers is not in the document]
Comment (AlbertoPdRF) [reply WcaOnRails/db/migrate/20180831164420_add_z1_and_z3_columns_to_competitions.rb]: The reason is that we don't know if past competitions applied this Regulation or not, so setting it to `NULL` is the way to say "we don't know".

I can resolve the conflict tomorrow!

**Comment by AlbertoPdRF** [comment]
The conflict should be resolved now.

**Pull request merged**